src/DataHandling/Logs/DataDogAudit.php

```diff
@@ -165,22 +271,242 @@ protected function removeColumnIfOnlyUnchanged($diffArray)
 
     /**
      * Method for getting key storing information about given on input column name with association. Can be override for customization needs.
-     * @param \DataDog\AuditBundle\Entity\AuditLog $currentVersion
-     * @return string name 
+     *
+     * @param AuditLog $currentVersion
+     *
+     * @return string name
      */
     protected function getColumnNameForAssociation(AuditLog $currentVersion)
     {
-        return $currentVersion->getTarget()->getTbl();
+        $joinTableName = $currentVersion->getTbl();
+        if (isset($this->cache['assocTable'][$joinTableName])) {
+            return $this->cache['assocTable'][$joinTableName];
+        }
+
+        foreach ($this->getEntitiesClassMetaData()->getAssociationMappings() as $assMapping) {
+            if (isset($assMapping['joinTable'])) {
+                $assJoinTable = $assMapping['joinTable']['name'];
+                $this->cache['assocTable'][$assJoinTable] = $assMapping['fieldName'];
+                if ($joinTableName === $assJoinTable) {
+                    return $assMapping['fieldName'];
+                }
+            } elseif (is_null($currentVersion->getTarget())
+                && $currentVersion->getSource()->getClass() === $assMapping['targetEntity']
+            ) { // ManyToOne mapping
+                $this->cache['assocTable'][$joinTableName] = $assMapping['fieldName'];
+
+                return $assMapping['fieldName'];
+            }
+        }
+
+        throw new \LogicException(sprintf('no association for %s found.', $joinTableName));
```

Comment:
It is called "Single Exit Point" rule. But, as written here https://www.tomdalling.com/blog/coding-tips/coding-tip-have-a-single-exit-point/ : "There is some controversy surrounding returning early versus having a single exit point, and both sides have decent arguments. However, it is my belief that robustness is more important than insignificant performance gains, and having a single exit point improves maintainability for any non-trivial function."

Reply:
disagree that single entry is more robust.
anyway

Reply:
I would write this method like that to follow "Single Exit Point" rule: 
`        $joinTableName = $currentVersion->getTbl();
		$returnValue = false;
		
        if (!isset($this->cache['assocTable'][$joinTableName])) {
            foreach ($this->getEntitiesClassMetaData()->getAssociationMappings() as $assMapping) {
				if (isset($assMapping['joinTable'])) {
					$assJoinTable = $assMapping['joinTable']['name'];
					$this->cache['assocTable'][$assJoinTable] = $assMapping['fieldName'];
					if ($joinTableName === $assJoinTable) {
						$returnValue = $assMapping['fieldName'];
					}
				} elseif (is_null($currentVersion->getTarget())
					&& $currentVersion->getSource()->getClass() === $assMapping['targetEntity']
				) { // ManyToOne mapping
					$this->cache['assocTable'][$joinTableName] = $assMapping['fieldName'];
					$returnValue = $assMapping['fieldName'];
				}
            }
        } else {
			$returnValue = $this->cache['assocTable'][$joinTableName];
		}
        
		if (!$returnValue) {
			throw new \LogicException(sprintf('no association for %s found.', $joinTableName));
		}
		
		return $returnValue;`
 In such case we don't have loss of performance.

Reply:
I would write this method like that to follow "Single Exit Point" rule: 
`        $joinTableName = $currentVersion->getTbl();
		$returnValue = false;
		
        if (!isset($this->cache['assocTable'][$joinTableName])) {
            foreach ($this->getEntitiesClassMetaData()->getAssociationMappings() as $assMapping) {
				if (isset($assMapping['joinTable'])) {
					$assJoinTable = $assMapping['joinTable']['name'];
					$this->cache['assocTable'][$assJoinTable] = $assMapping['fieldName'];
					if ($joinTableName === $assJoinTable) {
						$returnValue = $assMapping['fieldName'];
						break;
					}
				} elseif (is_null($currentVersion->getTarget())
					&& $currentVersion->getSource()->getClass() === $assMapping['targetEntity']
				) { // ManyToOne mapping
					$this->cache['assocTable'][$joinTableName] = $assMapping['fieldName'];
					$returnValue = $assMapping['fieldName'];
					break;
				}
            }
        } else {
			$returnValue = $this->cache['assocTable'][$joinTableName];
		}
        
		if (!$returnValue) {
			throw new \LogicException(sprintf('no association for %s found.', $joinTableName));
		}
		
		return $returnValue;`
 In such case we don't have loss of performance.